crate/operator/cratedb.py

```diff
@@ -109,11 +109,53 @@ async def get_healthiness(cursor: Cursor) -> int:
     return row and row[0]
 
 
+async def is_cluster_healthy(
+    connection_factory, expected_nodes: int, logger: logging.Logger
+) -> bool:
+    """
+    Check if a cluster is healthy.
+
+    The function checks for the cluster health using the `sys.health
+    <https://crate.io/docs/crate/reference/en/latest/admin/system-information.html#health>`_
+    table and the expected number of nodes in the cluster three times.
+
+    :param connection_factory: A callable that allows the operator to connect
+        to the database. We regularly need to reconnect to ensure the
+        connection wasn't closed because it was opened to a CrateDB node that
+        was shut down since the connection was opened.
+    :param expected_nodes: The number of nodes that make up a healthy cluster.
+    """
+    # We need to establish a new connection because the peer of a
+    # previous connection could have been shut down. And by
+    # re-establishing a connection for _each_ polling we can assert
+    # that the connection is open
+    async with connection_factory() as conn:
+        async with conn.cursor() as cursor:
+            logger.info("Waiting for cluster to get healthy again ...")
```

Comment:
For me, it looks like here we are waiting to get the results of a health check and not waiting until a cluster is healthy. At least if you just consider the scope of the function. 🤔